Makefile

```
@@ -67,3 +67,21 @@ help: Makefile
 	@echo ''
 	@echo 'Targets:'
 	@sed -n 's/^#?//p' $< | column -t -s ':' |  sort | sed -e 's/^/ /'
+
+test-coverage:
+	@if [ "$(word 2,$(MAKECMDGOALS))" = "" ]; then \
+		echo "Usage: make test-coverage <directory>"; \
+		echo "Example: make test-coverage ./miner"; \
+		echo "         make test-coverage ./..."; \
+		exit 1; \
+	fi
+	@PKG=$(word 2,$(MAKECMDGOALS)); \
+	if echo "$$PKG" | grep -q "^\./" ; then \
+		go test -coverprofile=coverage.out $$PKG; \
+	else \
+		go test -coverprofile=coverage.out ./$$PKG; \
+	fi
+	go tool cover -html=coverage.out -o coverage.html
+	open coverage.html  # macOS
+	sleep 0.5
+	rm coverage.out coverage.html
```

Maybe seaparating out cleaning the test coverage dir with a separate makefile command. Then call the cleanup at the start and end of the test-coverage command
```suggestion
COVER_DIR := .coverage
COVER_OUT := $(COVER_DIR)/coverage.out
COVER_HTML:= $(COVER_DIR)/coverage.html

test-coverage-clean:
	@rm -rf $(COVER_DIR)
```

---

But this means that the target will additionally need to have a mkdir for the .coverage directory, just to remove the directory in the end. Would simply creating the 2 files and deleting them in the same target be more straightforward?